engine/runner.ml

```diff
@@ -727,11 +755,19 @@ let main repo_dir nocache deploy =
   let%lwt new_configs = get_configs repo_dir in
   (*TODO Handle printing exceptions better, maybe use Fmt?*)
   let nodes = R.failwith_error_msg (parse_configs new_configs) in
+  let%lwt () =
+    if List.length target_nodes > 0 then
```

Comment:
This might be slightly clearer with pattern matching:
```ocaml
match target_nodes with
| [] -> Lwt_io.printl "Running all nodes due to no targetting being selected."
| _ -> Lwt_io.printl "Running only select nodes due to targetting."
```
In either case it's probably worth printing the nodes that will be run in the non-empty branch.